src/tbp/monty/frameworks/config_utils/make_dataset_configs.py

```diff
@@ -1102,3 +1104,276 @@ class FiveLMMountConfig:
+            identical to row 0. Otherwise, the array has `n_sms` rows. row 0 is
+            always centered at (0, 0, 0), and all other rows are offset relative to it.
+
+    """
+    assert n_sms > 0, "n_lms must be greater than 0"
+    assert delta > 0, "delta must be greater than 0"
+    assert order_by in ["spiral", "distance"], "order_by must be 'spiral' or 'distance'"
+
+    # Find smallest square grid size that can fit n_lms with odd-length sides.
+    n = 1
+    while n_sms > n**2:
+        n += 2
+
+    # Make coordinate grids, where the center is (0, 0).
+    pts = np.arange(-n // 2 + 1, n // 2 + 1)
+    x, y = np.meshgrid(pts, pts)
+    y = np.flipud(y)
+    i_mid = n // 2
+
+    if order_by == "distance":
+        dists = x**2 + y**2
+        unique_dists = np.sort(np.unique(dists))
+        assert unique_dists[0] == 0
+        indices = []
+        for i in range(len(unique_dists)):
+            u = unique_dists[i]
+            inds = np.argwhere(dists == u)
+            angles = np.arctan2(i_mid - inds[:, 1], inds[:, 0] - i_mid)
+            sorting_inds = np.argsort(angles)
+            inds = inds[sorting_inds]
+            indices.extend([row for row in inds])
+
+    elif order_by == "spiral":
+        indices = [(i_mid, i_mid)]
+
+        # Directions for moving in spiral: right, down, left, up
+        directions = [(0, 1), (1, 0), (0, -1), (-1, 0)]
+        current_dir = 0  # Start moving right
+        steps = 1  # How many steps to take in current direction
+        steps_taken = 0  # Steps taken in current direction
+        row, col = i_mid, i_mid  # Start at center
+
+        # Generate spiral pattern until we have enough points
+        while len(indices) < n**2:
+            # Move in current direction
+            row += directions[current_dir][0]
+            col += directions[current_dir][1]
+
+            # Add point if it's within bounds
+            if 0 <= row < n and 0 <= col < n:
+                indices.append((row, col))
+
+            steps_taken += 1
+
+            # Check if we need to change direction
+            if steps_taken == steps:
+                steps_taken = 0
+                current_dir = (current_dir + 1) % 4
+                # Increase steps every 2 direction changes (completing half circle)
+                if current_dir % 2 == 0:
+                    steps += 1
+
+    indices = np.array(indices)[:n_sms]
+
+    # Convert indices to locations in agent space.
+    positions = []
+    for idx in indices:
+        positions.append((x[idx[0], idx[1]] * delta, y[idx[0], idx[1]] * delta))
+    positions = np.array(positions)
+
+    # Add z-positions.
+    positions = np.hstack((positions, np.zeros([positions.shape[0], 1])))
+
+    # Optionally append entry for a view finder which is a duplicate of row zero.
+    # Should be (0, 0, 0).
+    if add_view_finder:
+        positions = np.vstack([positions, positions[0].reshape(1, -1)])
+
+    return positions
+
+
+def make_multi_lm_mount_config(
```

Comment:
maybe here again we make it a bit more specific and call it multi_sensor instead of multi_lm since the dataset args have nothing to do with lms

Reply:
What do you think of `multi_sm...` for these names?

In general, I went with `multi_lm` for everything across `make_dataset_args` and `config_args` since they all kinda basically belong together in a practical sense. That said, I went back and forth on it using `multi_lm` in `config_args` and `multi_sm` for `make_dataset_args`. It sounds like the preference is for either `multi_sensor` (or `multi_sm`, which I think is more consistent with `multi_lm` naming).

Reply:
Interesting, I hadn't considered multiple sensors connected to a single sensor module.